### DISCUSSION TITLE
Shovel with Filtering/Header Exchange based Shovel

### DISCUSSION THREAD

**Discussion by VipinEV** [rabbitmq/rabbitmq-server#8983] [Other] [answered]
### Is your feature request related to a problem? Please describe.

Hello Team,

Would it be good to have the below feature?
 
vhost1 > exchange1 > filter set of messages > vhost2 > exchange2
OR
vhost1 > header-exchange1 > header based filtering > vhost2 > queue2.

Do we have any workaround for this now?

### Describe the solution you'd like

This will help to

1. Use rabbitmq for broadcasting messages to multiple client in other vhosts with message filtering. 

### Describe alternatives you've considered

1. Declare a header exchange in vhost1.
2. Bind a queue - q1.
3. Create shovel from q1 to q2 (in vhost2)

### Additional context

The above approach ends up in creating more and more replication queues in vhost1 per each client.

**Comment by michaelklishin** [comment]
Using exchange to exchange bindings you can do just that on both source and destination ends.

**Comment by VipinEV** [comment]
@michaelklishin e2e binding possible between vhosts?

**Reply by michaelklishin** [reply, marked as answer]
No, by the definition of virtual hosts they are not. But R2-D2 providers filtering and Shovels can use a queue bound to a “filtered” destination exchange, as a source. Same idea can also be applied on the destination side.

We will not complicate this any further with Shovel-specific filtering.

Finally, Shovel can be used to move messages between virtual hosts in the same cluster.

**Reply by VipinEV** [reply]
Thanks, thats one of the alternative we tried ended up in creating multiple filtered source queue for multiple clients sitting in different vhost. I guess thats the only solution available at the moment.
